README.md

```diff
@@ -158,6 +166,80 @@ The emitter has to be provided when creating an instance of `XRayRecorder`:
 let recorder = XRayRecorder(emitter: XRayNoOpEmitter())
 ```
 
+### Context propagation
+
+Unlike in other X-Ray SDKs, `XRayRecorder` in AWS X-Ray SDK for Swift does not expose (thread local) current `Segment`. 
+
+The context, in its broader meaning that is including but not limited to trace context, should be passed explicitly in `BaggageContext`:
+
+```swift
+// the baggage should contain trace context
+var baggage: BaggageContext
+
+let segment = recorder.beginSegment(name: "Segment 1", baggage: baggage)
```

Comment:
I still suggest this should accept a `context: BaggageContextCarrier`, this way existing frameworks can pass their `tracer.startSpan(..., context: request)` etc. 

There's a ticket about naming which you might find interesting as well https://github.com/slashmo/gsoc-swift-baggage-context/issues/23 that'd hopefully be the last renaming we have to tackle.

Reply:
@ktoso thank you for the comments, much appreciated

I will look into it but will wait till https://github.com/slashmo/gsoc-swift-baggage-context/issues/23 is resolved as it may bring in more changes

Also I want to have a closer look at  `BaggageLogging`.
My, not so urgent, goal is have a nice integration with (AWS) CloudWatch ServiceLens (per https://github.com/pokryfka/aws-xray-sdk-swift/issues/27),
which groups X-Ray traces with CloudWatch logs.

Reply:
Yes makes sense to wait out #23 a little bit.

Awesome that you're also looking at logs, that's indeed a goal we have as well so let's see if we need to make anything special there. Today's implementations make _all_ baggage items available to a logger when logging; So if that's too much we can either invest in better logger backends that allow filtering what they'd print or allowing the BaggageLogHandler to be smarter about which values it exposes by allowing a BaggageKey to be configured with some "hide from logging" etc... Open to ideas, but the simple thing should work today -- i.e. simply all values in baggage would be logged.